src/foam/dao/RoutingJDAO.js

```diff
@@ -49,37 +50,18 @@ new Thread() {
   ],
 
   methods: [
-    {
-      name: 'find_',
-      flags: null,
-      javaCode: `
-if ( x.get("replayingJournal") == getJournal() ) {
-  return getDelegate().find_(x, id);
-} else {
-  return super.find_(x, id);
-}
-      `
-    },
     {
       name: 'put_',
       flags: null,
       javaCode: `
-if ( x.get("replayingJournal") == getJournal() ) {
-  return getDelegate().put_(x, obj);
-} else {
-  return super.put_(x.put("service", getService()), obj);
-}
+return super.put_(x.put("service", getService()), obj);
```

Comment:
Rather than create a new context on every put/remove in the system (could be slow), let's decorate the journal that each routingJDAO has to know the service name and pass that along to the underlying routing journal as a plain old argument.